### ISSUE TITLE
Show warning if commit message is empty

### ISSUE THREAD

**Issue by cvetan** [JohnstonCode/svn-scm#240]
I find behaviour of the extension very weird and inconsistent. I didn't commit all the changes i wanted serveral times, before i figured out how to do things.

For instance for deleted and newly added files, you have to be explicit, you click on them in vcs view and add/remove them. Then you can commit them from command pallete for instance, aka they are "staged" for commit so to speak.

But for modified files you have to select them all and then "commit selected".

But if you do that for deleted or newly added files, before you click add/remove you get nothing, nothing happens. 

This makes a lot of confusion, at least for me. You have to treat newly added, and deleted files in one way, but modified files in totally different way. I would like to hear from you, and your oppinions and also is this unconsistent behaviour intentional.

**Comment by petrmusil** [comment]
I have similar feeling. It almost look like a bug.

The commit is critical action and, in my opinion, it has to be absolutely clear how things works.

Maybe the TortoiseSVN GUI can be used for inspiration how to solve this.

**Comment by JohnstonCode** [comment]
The commit selected option should not be available on from the command pallet. I will remove this.

The two options for committing files should be either the commit bar in the SCM view or ctrl-clicking and right-clicking and choosing commit selected.

**Comment by petrmusil** [comment]
@JohnstonCode Great, let`s try this for future release.
I am available for feedback on changes anytime.
Really appreciate you work!

**Comment by cvetan** [comment]
For me the problem was not command pallete that much, as the thing for the modified files you need to "hold" them and then click commit these files. As for deleted/added files you just add them/delete them with respected options you then don't have to hold them in commit, they will be commited. I think this should be option for modified files as well, or even better those file should be commited by default, if are in the main changes list, and not any other changelist. Like that it would be much more consistent. You don't have to explicitly "add" modified files.

**Comment by petrmusil** [comment]
Exactly. We just need need to easily commit files with some behavior that is intuitive and matches most common use-cases.
One thing is to show, what files are being affected by commit.
Second, there should be some defaults for each "status" (changed file, new file, missing file,... etc.). The defaults can look something like this (take it as an example):
1. changed file/folder: commit
2. new file/folder: add, commit
3. missing file/folder: commit deletion
4. renamed file/folder: treat as change (?)
+ rule to exclude ignored files

I believe this is more complex, and I am not so deep in this.

Also, I found very confusing that Ctrl+click select file, but Ctrl+Shift+A (to select file/s) behave in a different way?

**Comment by JohnstonCode** [comment]
I must be missing something. I honestly done understand what issue you both are having committing files.

**Comment by petrmusil** [comment]
@JohnstonCode This doesn't work for me: "Either enter something in the commit message box and then everything listed as changes is committed or individually select files and commit them."

Assuming I have one or more changed files:
1) Hitting commit icon (check icon): does nothing (neither prompt commit message input, nor commit is done)
2) Selecting files in Source control pane using Ctrl+A  and using command "Commit selected" (from command pallet or context menu above source control panel (right-click) works as mentioned in previous posts

![image](https://user-images.githubusercontent.com/10118027/38487771-92a58928-3be1-11e8-91cc-d97cad9a6192.png)

VS Code:

Version 1.21.1
Commit 79b44aa704ce542d8ca4a3cc44cfca566e7720f1
Date 2018-03-14T14:46:47.128Z
Shell 1.7.9
Renderer 58.0.3029.110
Node 7.9.0
Architecture x64

**Comment by petrmusil** [comment]
OMG now I read your reply one more time and I got where the problem is.

The grey box in Source control panel is for commit message AND HAS TO BE FILLED.

OK the I withdraw all my previous posts.

But seriously... this is not user friendly at all:

![image](https://user-images.githubusercontent.com/10118027/38488033-6f2c6f06-3be2-11e8-824c-4a53b93aeb6c.png)

Please consider providing feedback to the user in this situation. A message and highlighting input as required should be a good practice to follow.

To sum it up:
1) As an TortoiseSVN GUI user and also from a context of Command palet I expected to forts initiate command and then enter the commit message
2) I did not receive any feedback that commit message must be provided before "commit all" command right from GUI
3) As a consequence I supposed I have do to some sort of selection of files to commit files

**Comment by JohnstonCode** [comment]
@petrmusil aaah i see now!!!

Yes, there should be some feedback there.

**Comment by cvetan** [comment]
I think that main commit, from source control view now works as it should. All the changes that were listed in "changes" were commited. When i reported this, that was not the case. With this warnings for empty commit message, i think that would be much better. Or even better commit without message. I do that a lot of times, when there isn't significant change so i don't even enter message.

**Comment by JohnstonCode** [comment]
I was thinking about just adding a confirm prompt like "Commit without message"?

**Comment by edgardmessias** [comment]
My suggestion: If message is empty, show confirmation "Commit Message is Empty! Continue?"

**Comment by petrmusil** [comment]
As I stated in one of my previous post -- to make commit message optional is **preferred** (I can think about many use-cases when this is absolutely OK, also SVN let you do this no problem).

Suggestion from @edgardmessias makes this solution perfect imo.

**Comment by JohnstonCode** [comment]
@edgardmessias sure

**Comment by edgardmessias** [comment]
Done with #245
